src/backend/tensilica-g3.rkt

```diff
@@ -48,27 +48,33 @@
 
   do-read)
 
+; Track the C type of each id
+(define type-tracker (make-hash))
+(define (type-set id ty)
+  (hash-set! type-tracker (id->string id) ty))
+(define (type-ref id)
+ (hash-ref type-tracker (id->string id)))
```

Comment:
I'm a fan of wrapping state and returning the lambdas to access them instead of having it present in the global environment.